gf_cpu.c: fix pclmul detection and use gcc/clang provided bit defs

dx9 opened the pull request.

tchaikov commented:
@dx9 could you also send a pull request to the upstream of gf-complete at http://lab.jerasure.org/jerasure/gf-complete?

dx9 committed:
gf_cpu.c: fix pclmul detection and add portable cpuid feature bit defs

Correct invalid check for pclmul support. Was checking SSE3 (1 << 0) vs. PCLMUL (1 << 1).

Fixes: http://tracker.ceph.com/issues/18092
Signed-off-by: John Coyle <dx9err@gmail.com>

dx9 commented:
yep http://lab.jerasure.org/jerasure/gf-complete/merge_requests/20
